train.c

```diff
@@ -0,0 +1,90 @@
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
+#include <float.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include "agents/temporal_difference.h"
+#include "game.h"
+
+static double epsilon = 0.3;
+#define NUM_ITERATION 10000
+#define RAND_UNIFORM ((float) rand() / (float) RAND_MAX)
+static unsigned int N_STATES = 1;
+static td_agent_t agent[2];
+
+int get_action_epsilon_greedy(char *table, td_agent_t *agent)
+{
+    int move_cnt = 0;
+    int *available_moves = get_available_moves(table, &move_cnt);
```

Comment:
I didn't get how to replace `get_available_moves` with `list_for_each` style macro here, is there another way to get a random move by iterating the board straightly?

Reply:
I'm sorry that I didn't think clearly. We can't obtain the random space straightly. We need to complete it in two steps:
1. get a random count i between [0, the number of spaces).
2. get the board index of the ith space.

This method has the same time complexity as the original one but is not very elegant.

Reply:
I've move `get_available_moves` into train.c and rewrite it with `for_each_empty_grid`. Thanks for your advice!.

Reply:
> I'm sorry that I didn't think clearly. We can't obtain the random space straightly. We need to complete it in three steps:
> 
> 1. Get the number of spaces n by `for_each_empty_grid`.
> 2. Get a random count i between [0, n).
> 3. Get the board index of the ith space by `for_each_empty_grid`.
> 
> This method has the same time complexity as the original one but is not very elegant.

However, I still prefer my proposed method to get a random empty cell because it's in-place.
Additionally, could you also replace the usage of `available_moves` function in `negamax.c` and move `for_each_empty_grid` to `game.h`.
And is `for_each_empty_cell` a good name?